### DISCUSSION TITLE
How to manually trigger complete in select mode?

### DISCUSSION THREAD

**Discussion by weilbith** [hrsh7th/nvim-cmp#1755] [Q&A]
Hey :wave:

I try to do a simple mapping: `:smap <C-r> <cmd>lua require('cmp').complete()<CR>`. But it does not open the completion menu. The same for `imap` just works fine. 

What am I missing?
I tried to search in the issues and discussions for help. Sorry I missed something. The term `select` is overloaded for this plugin. 🙈

**Comment by fitrh** [comment]
I think that's something that isn't currently possible

https://github.com/hrsh7th/nvim-cmp/blob/main/lua/cmp/core.lua#L275-L277

https://github.com/hrsh7th/nvim-cmp/blob/main/lua/cmp/utils/api.lua#L35-L38

**Reply by weilbith** [reply]
Oh, thanks. That are really good links. But at the same time very frustrating. Wonder if I could overwrite that. Or if it would just not work. 🤔

**Reply by weilbith** [reply]
Okay, so manipulating this function locally does the job for me. So it actually works mostly fine. I can select entries and confirm them. I don't get ghost text, but that is totally fine.
Just trying to figure out how to do this without a local source code manipulation.

**Reply by weilbith** [reply]
@hrsh7th may I ask whats the issue/reasoning that completion should be not allowed in select mode? 🙂

**Reply by Shougo** [reply]
Hm...  I think you can move into insert mode from select mode.
Why you need to open completion window directly?

**Reply by Shougo** [reply]
I think it works.

```vim
smap <C-r> <C-o>"_di<cmd>lua require('cmp').complete()<CR>
```

**Reply by weilbith** [reply]
Thanks for the input!
I'm trying to improve the experience of the [cmp-luasnip-choice](https://github.com/doxnit/cmp-luasnip-choice) plugin. It works kinda okay. But it doesn't work for choice nodes where you end up in select mode. 
I'm trying to think if the insert mode "work around" works well for this scenario. 🤔

**Reply by Shougo** [reply]
Hm...  It seems limited scenario.
